File: src/cli-sample/PrintDeviceNames.cs
using System;

namespace Xt {

    public static class PrintDeviceNames {

        [STAThread]
        public static void Main(string[] args) {

            using (XtAudio audio = new XtAudio(null, IntPtr.Zero, null, null)) {
                for (int s = 0; s < XtAudio.GetServiceCount(); s++) {
                    XtService service = XtAudio.GetServiceByIndex(s);
                    for (int d = 0; d < service.GetDeviceCount(); d++)
                        using (XtDevice device = service.OpenDevice(d)) {
                            Console.WriteLine(service.GetName() + ": " + device.GetName());
                        }
                }
            }
        }
    }
}

File: src/java-sample/com/xtaudio/xt/PrintDeviceNames.java
package com.xtaudio.xt;

import com.sun.jna.Pointer;

public class PrintDeviceNames {

    public static void main(String[] args) {
        
        try (XtAudio audio = new XtAudio(null, Pointer.NULL, null, null)) {
            for (int s = 0; s < XtAudio.getServiceCount(); s++) {
                XtService service = XtAudio.getServiceByIndex(s);
                for (int d = 0; d < service.getDeviceCount(); d++)
                    try (XtDevice device = service.openDevice(d)) {
                        System.out.println(service.getName() + ": " + device.getName());
                    }
            }
        }
    }
}
